apps/src/code-studio/progress.js

```diff
@@ -284,10 +289,15 @@ function initializeStoreWithProgress(
       scriptDescription: scriptData.description,
       betaTitle: scriptData.beta_title,
       courseId: scriptData.course_id,
-      isFullProgress: isFullProgress
+      isFullProgress: isFullProgress,
+      onLessonExtras: onLessonExtras
     })
   );
 
+  if (scriptData.lessons.length === 1) {
+    store.dispatch(setCurrentStageId(scriptData.lessons[0].id));
```

Comment:
Why do we only set this if a lesson has length of 1?

Reply:
Our logic is that if we only have one lesson, we auto-select it. I added a comment.